option.go

```diff
@@ -184,6 +196,44 @@ func FloatOption(names ...string) Option {
 func StringOption(names ...string) Option {
 	return NewOption(String, names...)
 }
+
+// StringsOption is a command option that can handle a slice of strings
 func StringsOption(names ...string) Option {
-	return NewOption(Strings, names...)
+	return DelimitedStringsOption("", names...)
+}
+
+// DelimitedStringsOption like StringsOption is a command option that can handle a slice of strings.
+// However, DelimitedStringsOption will automatically break up the associated CLI inputs based on the delimiter.
+// For example, instead of passing `command --option=val1 --option=val2` you can pass `command --option=val1,val2` or
+// even `command --option=val1,val2 --option=val3,val4`.
+//
+// A delimiter of "" means that no delimiter is used
+func DelimitedStringsOption(delimiter string, names ...string) Option {
```

Comment:
Went with two different names here because just adding another string option called delimiter in front will not cause a compile error for any users of the current non-delimiter StringsOption which seems useful (at least while drafting the PR if not further).

Options I can see so far include:
- Two options as the PR has now
- Just one option, but making sure to fix all the cases in this library and go-ipfs (seems error prone, but doable)
- Use one option, but instead of a string delimiter pass in a non-convertable type or option so that we get a compile error